2022/incoming/the-delivery-map.pod

```diff
@@ -0,0 +1,149 @@
+Author: Thibault DUPONCHELLE <thibault.duponchelle@gmail.com>
+Title: The delivery map
+Topic: GraphViz2
+
+=encoding utf8
+
+=head1 The delivery map
+
+Years after years, Santa is becoming more "absent minded" and he tends to forget little things like the keys of the sled or he forgets to put his hat.
+
+To give you an idea, in 2021, Santa forgot to deliver the Family Jones in Birmingham... and had to come back later to arrange this miss!
+
+This kind of mistake takes time and is tiring for Santa and his right-hand elf, worried about this, just came with a solution. 
+
+He started to design "TODO-Maps" for his boss.
+
+For this purpose, he decided to use L<GraphViz2> Perl module, to build a graph of families to deliver.
+
+L<GraphViz|https://graphviz.org/> is a well known graph visualization software and is easy to use, and the elf was able to come with a first graph in matter of minutes:
+
+=begin perl
+
+#!/usr/bin/env perl
+
+use GraphViz2;
+
+my $graph = GraphViz2 -> new(graph  => {rankdir => 'LR'});
+
+$graph -> add_edge(from =>  "Family Anderson", to =>  "Family Jones");
+$graph -> add_edge(from =>  "Family Jones", to =>  "Family Taylor");
+$graph -> add_edge(from =>  "Family Taylor", to =>  "Family Williams");
+
+$graph->run(format => 'svg', output_file => 'simple.svg');
+
+=end perl
+
+And got this nice result:
+
+=begin html
+
+<center><img src="../share/static/simple.svg" alt="A simple delivery map"></center>
+
+=end html
+
+=head2 Make it a bit better
+
+It was already a nice help for Santa, but the elf wanted to make it a bit better.
+
+First, to materialize the direction using C<< directed => 1 >> and change the node shape to house C<< shape => "house" >> because after all, these families live in houses!
+
+=begin perl
+
+#!/usr/bin/env perl
+
+use GraphViz2;
+
+my $graph = GraphViz2 -> new(
+	global => {name => 'Birmingham', directed => 1},
+	graph  => {rankdir => 'LR'},
+);
+
+$graph->default_node(
+	shape    => "house",
+	width    => "2",
+	height   => "2",
+);
+
+$graph -> add_edge(from =>  "Family Anderson", to =>  "Family Jones");
+$graph -> add_edge(from =>  "Family Jones", to =>  "Family Taylor");
+$graph -> add_edge(from =>  "Family Taylor", to =>  "Family Williams");
+
+$graph->run(format => 'svg', output_file => 'better.svg');
+
+=end perl
+
+And got this result:
+
+=begin html
+
+<center><img src="../share/static/better.svg" alt="A better delivery map"></center>
+
+=end html
+
+=head2 The full procedure for Santa
+
+Santa was already happy with this map but if was thinking about disconnecting a bit more. 
+
+So if it would be possible to make it appear also the "preparation" steps, it would be perfect for Santa!
```

Comment:
If the map could also display the chores he would do to get ready for the day, it would be perfect for Santa.

Reply:
Edited